ioctl/test_ioctl.c

```diff
@@ -0,0 +1,405 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    Phoenix-RTOS standard library functions tests
+ *    HEADER:
+ *    - ioctl.h
+ *    TESTED:
+ *    - ioctl()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <phoenix/ioctl.h>
+#include <sys/msg.h>
+#include <signal.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <posix/utils.h>
+#include <unity_fixture.h>
+
+
+static uint32_t port;
+static pthread_t tid;
+static int ret, fd, dd = 5;
+static int loop_flag = 1;
```

Comment:
needs to be volatile due to being changed from other thread. Probably you could define special ioctl (`TEST_IOCTL_QUIT`)  which would exit the loop without passing the state via global variable

Reply:
IMHO test case is supposed to test something and if I am able to exit loop via global variable in controled and safe way I would stick with it.